api/src/db/migrate.ts

```diff
@@ -0,0 +1,259 @@
+import Session from '../models/session';
+import { db } from '.';
+import { planner, report, review, savedCourse, session, transferredCourse, user, vote } from './schema';
+import Review from '../models/review';
+import Roadmap from '../models/roadmap';
+import Preference from '../models/preference';
+import Report from '../models/report';
+import Vote from '../models/vote';
+import { anonymousName, MongoRoadmap, SavedPlannerData, SavedPlannerYearData } from '@peterportal/types';
+import mongoose from 'mongoose';
+
+const uri = process.env.MONGO_URL;
+const conn = await mongoose.connect(uri!, {
+  dbName: 'peterPortalDevDB',
+  serverSelectionTimeoutMS: 5000,
+});
+
+// clear first
+await Promise.all([
+  db.delete(planner),
+  db.delete(vote),
+  db.delete(report),
+  db.delete(review),
+  db.delete(savedCourse),
+  db.delete(transferredCourse),
+  db.delete(session),
+  db.delete(review),
+  db.delete(user),
+]);
+
+// get all possible users from sessions, then reviews, then roadmaps, then votes
+const sessions = await Session.find();
+// need to filter out bad sessions (never signed in with google, just clicked log in)
+const sessionsUserRecords = sessions
+  .filter((session) => session.session?.passport?.user?.id != null)
+  .map((session) => ({
```

Comment:
I know it's nice to be able to directly return the object, but at the same time repeating `sessions.session!.passport!.user!` is a bit clunky

If you _really_ want to, you could also do `.map(({ session: { session: { passport: { user } } }) => ({...})` but that's equally as confusing

Reply:
I think it'd be best to just do something like this:

```ts
.map(session => {
  const { id, name, email, picture } = session.session!.passport!.user!
  return { googleId: id, name, email, picture }
})
```

Reply:
i don't really care about the code in this file because i'm not merging it, this just for a one-time use of migrating everything over